engines/native/sandbox.go

```diff
@@ -47,11 +47,28 @@ func newSandbox(b *sandboxBuilder) (*sandbox, error) {
 		}
 	}
 
-	// Create temporary user account
-	user, err := system.CreateUser(homeFolder.Path(), b.engine.groups)
-	if err != nil {
-		homeFolder.Remove() // best-effort clean-up this is a fatal error
-		return nil, fmt.Errorf("Failed to create temporary system user, error: %s", err)
+	var user *system.User
+	var username string
+
+	if b.engine.config.CreateUser {
+		// Create temporary user account
+		user, err = system.CreateUser(homeFolder.Path(), b.engine.groups)
+		if err != nil {
+			homeFolder.Remove() // best-effort clean-up this is a fatal error
+			return nil, fmt.Errorf("Failed to create temporary system user, error: %s", err)
+		}
+		username = user.Name()
+	} else {
+		var curUser *system.User
+
+		user = nil
+
+		curUser, err = system.CurrentUser()
+		if err != nil {
+			return nil, err
```

Comment:
Don't forget to clean up `homeFolder.Remove() // best-effort clean-up this is a fatal error` as in the error handling for if-branch.